doc/source/main-documentation/qibolab.rst

```diff
@@ -0,0 +1,611 @@
+
+It is possible to define and executes multiple sweepers at the same time.
+For example:
+
+.. code-block:: python
+
+    sequence = PulseSequence()
+
+    sequence.add(my_platform.create_RX_pulse())
+    sequence.add(my_platform.create_MZ_pulse(start=sequence[0].finish))
+
+    sweeper_freq = Sweeper(
+        parameter=Parameter.frequency,
+        values=np.arange(-100_000, +100_000, 10_000),
+        pulses=[sequence[0]],
+        type=SweeperType.OFFSET,
+    )
+    sweeper_amp = Sweeper(
+        parameter=Parameter.amplitude,
+        values=np.arange(0, 1.5, 0.1),
+        pulses=[sequence[0]],
+        type=SweeperType.FACTOR,
+    )
+
+    results = my_platform.sweep(sequence, options, sweeper_freq, sweeper_amp)
+
+Let's say that the RX pulse has, from the runcard, a frequency of 4.5 GHz and an amplitude of 0.3, the parameter space probed will be:
+
+- amplitudes: [0, 0.03, 0.06, 0.09, 0.12, ..., 0.39, 0.42]
+- frequencies: [4.4999, 4.49991, 4.49992, ...., 4.50008, 4.50009] (GHz)
+
+.. warning::
+
+   Different control devices may have different limitations on the sweepers.
+   It is possible that the sweeper will raise an error, if not supported, or that it will be automatically converted as a list of pulse sequences to perform sequentially.
+
+Execution Parameters
+--------------------
+
+In the course of several examples, you've encountered the ``options`` argument in function calls like:
+
+.. code-block:: python
+
+   res = my_platform.execute_pulse_sequence(..., options=options)
+   res = my_platform.sweep(..., options=options)
+
+Let's now delve into the details of the ``options`` parameter and understand its components.
+
+The ``options`` parameter, represented by the :class:`qibolab.execution_parameters.ExecutionParameters` class, is a vital element for every hardware execution. It encompasses essential information that tailors the execution to specific requirements:
+
+- ``nshots``: Specifies the number of experiment repetitions.
+- ``relaxation_time``: Introduces a wait time between repetitions, measured in nanoseconds (ns).
+- ``fast_reset``: Enables or disables fast reset functionality, if supported; raises an error if not supported.
+- ``acquisition_type``: Determines the acquisition mode for results.
+- ``averaging_mode``: Defines the mode for result averaging.
+
+The first three parameters are straightforward in their purpose. However, let's take a closer look at the last two parameters.
+
+Supported acquisition types, accessible via the :class:`qibolab.execution_parameters.AcquisitionType` enumeration, include:
+
+- Discrimination: Distinguishes states based on acquired voltages.
+- Integration: Returns demodulated and integrated waveforms.
+- Raw: Offers demodulated, yet unintegrated waveforms.
+- Spectroscopy: Designed specifically for spectroscopy applications, primarily supported by Zurich Instruments devices.
```

Comment:
Shall we mention "Spectroscopy" here? I would skip since it is not relevant for any instrument other than Zurich and I am not even sure if the code supports it.